components/datadog/agent/helm.go

```diff
@@ -243,3 +255,22 @@ func buildWindowsHelmValues(installName string, agentImagePath, agentImageTag, _
 		},
 	}
 }
+
+func configureFakeintake(values pulumi.Map, fakeintake *ddfakeintake.ConnectionExporter) {
+	additionalEndpointsEnvVar := pulumi.MapArray{
+		pulumi.Map{
+			"name":  pulumi.String("DD_ADDITIONAL_ENDPOINTS"),
+			"value": pulumi.Sprintf("{\"http://%s\": [\"FAKEAPIKEY\"]}", fakeintake.Host),
+		},
+		pulumi.Map{
+			"name":  pulumi.String("DD_LOGS_CONFIG_ADDITIONAL_ENDPOINTS"),
+			"value": pulumi.Sprintf("[{\"host\": \"%s\", \"port\": 80, \"is_reliable\": true, \"usessl\": false}]", fakeintake.Host),
+		},
+	}
+
+	if values["datadog"].(pulumi.Map)["env"] == nil {
+		values["datadog"].(pulumi.Map)["env"] = additionalEndpointsEnvVar
+	} else {
+		values["datadog"].(pulumi.Map)["env"] = append(values["datadog"].(pulumi.Map)["env"].(pulumi.MapArray), additionalEndpointsEnvVar...)
```

Comment:
Is there any chance that `values["datadog"].(pulumi.Map)["env"]` failed? If yes you should handle it.

Reply:
The `values` manipulated here has been pre-filled by either `buildLinuxHelmValues` or `buildWindowsHelmValues` which are both filling `values` with:
```go
	return pulumi.Map{
		"datadog": pulumi.Map{
…
```

So, `values["datadog"]` cannot be `nil`;
`values["datadog"]` cannot be anything else than a `pulumi.Map`;
and we handle both cases where `env` key is present or not in `values["datadog"].(pulumi.Map)`
So, I think we’re fine because of how `values` is filled before the call to this function.